pkg/reconciler/instances/eventing/preaction/removenatsoperatorstep.go

```diff
@@ -0,0 +1,129 @@
+	kubeClientProvider
+}
+
+func newRemoveNatsOperatorStep() *removeNatsOperatorStep {
+	return &removeNatsOperatorStep{
+		kubeClientProvider: defaultKubeClientProvider,
+	}
+}
+
+func defaultKubeClientProvider(context *service.ActionContext, logger *zap.SugaredLogger) (kubernetes.Client, error) {
+	kubeClient, err := kubernetes.NewKubernetesClient(context.Task.Kubeconfig, logger, &kubernetes.Config{
+		ProgressInterval: progressTrackerInterval,
+		ProgressTimeout:  progressTrackerTimeout,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return kubeClient, nil
+}
+
+func (r *removeNatsOperatorStep) Execute(context *service.ActionContext, logger *zap.SugaredLogger) error {
+	// todo skip the step for kyma 2.x+ version for that use the check which will be implemented as a follow-up for https://github.com/kyma-incubator/reconciler/issues/334
+	// no kyma 2.x+ clusters contain nats-operator resources
+	//clusterVersion, err := cluster.Version()
+	//if err != nil {
+	//	return err
+	//}
+	//if clusterVersion > 1 {
+	//	logger.With(log.KeyReason, "NATS-operator resources do not exist on clusters with kyma 2.x+ version").Info("Step skipped")
+	//	return nil
+	//}
+
+	kubeClient, err := r.kubeClientProvider(context, logger)
+
+	if err != nil {
+		return err
+	}
+
+	// decorate logger
+	logger = logger.With(log.KeyStep, removeNatsOperatorStepName)
+
+	err = r.removeNatsOperatorResources(context, kubeClient, logger)
+	if err != nil {
+		return err
+	}
+
+	err = r.removeNatsOperatorCRDs(kubeClient, logger)
+	return err
+}
+
+func (r *removeNatsOperatorStep) removeNatsOperatorResources(context *service.ActionContext, kubeClient kubernetes.Client, logger *zap.SugaredLogger) error {
+	// get charts from the version 1.2.x, where the NATS-operator resources still exist
+	comp := GetResourcesFromVersion(natsOperatorLastVersion, natsSubChartPath)
+
+	manifest, err := context.ChartProvider.RenderManifest(comp)
+	if err != nil {
+		return err
+	}
+
+	// set the right eventing name, which went lost after rendering
+	manifest.Manifest = strings.ReplaceAll(manifest.Manifest, natsSubChartPath, eventingNats)
+
+	logger.Info("Removing nats-operator chart resources")
+	// remove all the existing nats-operator resources, installed via charts
+	_, err = kubeClient.Delete(context.Context, manifest.Manifest, namespace)
+	return err
+}
+
+// delete the leftover CRDs, which were outside of charts
+func (r *removeNatsOperatorStep) removeNatsOperatorCRDs(kubeClient kubernetes.Client, logger *zap.SugaredLogger) error {
+	logger.Info("Removing nats-operator CRDs")
+	for _, crdName := range natsOperatorCRDsToDelete {
+		_, err := kubeClient.DeleteResource(crdPlural, crdName, namespace)
+		if err != nil && !errors.IsNotFound(err) {
+			logger.Errorf("Failed to delete the nats-operator CRDs, name='%s', namespace='%s': %s", crdName, namespace, err)
+			return err
+		}
+		if err == nil {
+			logger.Debugf("Deleted %s CRD from %s namespace", crdName, namespace)
+		}
```

Comment:
Improving Evening reconciler logs should be done as part of https://github.com/kyma-incubator/reconciler/issues/460.